pio/src/iSpindel.cpp

```diff
@@ -837,60 +837,62 @@ void sleepManager()
   }
 }
 
-void requestTemp()
-{
-  if (!DSreqTime)
-  {
-    DS18B20.requestTemperatures();
-    DSreqTime = millis();
-  }
-}
-
 void initDS18B20()
 {
-  if (my_OWpin == -1)
+  //Try using old pin if set
+  if (my_OWpin != -1)
   {
-    my_OWpin = detectTempSensor(OW_PINS);
-    if (my_OWpin == -1)
+    CONSOLE(F("TempInit: scanning for saved OW device on pin: "));
+    CONSOLELN(my_OWpin);
+    oneWire = new OneWire(my_OWpin);
+    DS18B20 = DallasTemperature(oneWire);
+    if (!DS18B20.getAddress(tempDeviceAddress, 0))
     {
-      CONSOLELN(F("ERROR: cannot find a OneWire Temperature Sensor!"));
-      return;
+      CONSOLE(F("TempInit: not found on: "));
+      CONSOLELN(my_OWpin);
+      CONSOLELN(F(" -> Reset my_OWpin "));
+      my_OWpin = -1;
     }
   }
-  pinMode(my_OWpin, OUTPUT);
-  digitalWrite(my_OWpin, LOW);
-  delay(100);
-  oneWire = new OneWire(my_OWpin);
-  DS18B20 = DallasTemperature(oneWire);
-  DS18B20.begin();
-
-  bool device = DS18B20.getAddress(tempDeviceAddress, 0);
-  if (!device)
+  //Find pin
+  if (my_OWpin == -1)
   {
-    my_OWpin = detectTempSensor(OW_PINS);
-    if (my_OWpin == -1)
+    for (uint8_t p = 0; p < sizeof(OW_PINS); p++)
     {
-      CONSOLELN(F("ERROR: cannot find a OneWire Temperature Sensor!"));
-      return;
-    }
-    else
-    {
-      delete oneWire;
-      oneWire = new OneWire(my_OWpin);
+      const byte pin = OW_PINS[p];
+      CONSOLE(F("TempInit: scanning for OW device on pin: "));
+      CONSOLELN(pin);
+      oneWire = new OneWire(pin);
       DS18B20 = DallasTemperature(oneWire);
-      DS18B20.begin();
-      DS18B20.getAddress(tempDeviceAddress, 0);
+
+      if (!DS18B20.getAddress(tempDeviceAddress, 0))
+      {
+        CONSOLELN(F("Tempinit: No OneWire sensor found"));
+        break;
```

Comment:
I was wrong with **break** here. Here should be continue and after `my_OWpin = pin;` should be break.